awx_collection/tools/roles/template_galaxy/tasks/main.yml

```diff
@@ -0,0 +1,38 @@
+---
+- name: Set the collection version in the tower_api.py file
+  replace:
+    path: "{{ collection_path }}/plugins/module_utils/tower_api.py"
+    regexp: '^    _COLLECTION_VERSION =.*'
+    replace: '    _COLLECTION_VERSION = "{{ collection_version }}"'
```

Comment:
This is going to get a little annoying over time. The development flow is that `make install_collection` is ran before many other important development items can be ran. Yes, there are other replacements made in these tasks, but for `awx.awx`, those are all no-ops. Not this. If you run this you get a diff in your source tree like

```diff
diff --git a/awx_collection/plugins/module_utils/tower_api.py b/awx_collection/plugins/module_utils/tower_api.py
index dc0e14c915..2bd98c1337 100644
--- a/awx_collection/plugins/module_utils/tower_api.py
+++ b/awx_collection/plugins/module_utils/tower_api.py
@@ -33,7 +33,7 @@ class ItemNotDefined(Exception):
 
 class TowerModule(AnsibleModule):
     # This gets set by the make process so whatever is in here is irrelevant
-    _COLLECTION_VERSION = "11.0.0"
+    _COLLECTION_VERSION = "11.2.0"
     _COLLECTION_TYPE = "awx"
     # This maps the collections type (awx/tower) to the values returned by the API
     # Those values can be found in awx/api/generics.py line 204
```

Also, if possible, I would prefer a more static replacement, as opposed to wild cards.

What I'm getting to, I'd prefer to hard code a version like

```python
    _COLLECTION_VERSION = "devel"
```

And use a flag to avoid replacing that, which can be set in `make collection_install`. The actual release pipelines will not care.

I also like that there is _a_ version of the collection which corresponds to the actual source. Yeah, we're doing a lot of build-time replacements, but I would like a sane way of behaving without those.

Reply:
For the version being coded in, I agree it's a pain because of the source modification. Initially I was hoping that there was a way to get the version of the collection directly from Ansible but I couldn't find anything. Another avenue I explored was trying to have a "version file" somewhere in the collection under a gitignore that the tower_api could load version info out of. Unfortunately, I was having problems locating the file but maybe we can explore that more.

Regarding the wild card replacement I did that somewhat intentionally because of the fact that running `make install_collection` changes  the source. I was attempting to compensate for the case where someone pushes a commit with the value of _COLLECTION_VERSION changed. In the scenario with devel if a PR reviewer misses that line being changed things will get messed up. With the wildcard replacement we don't have that issue.